mobile/lib/components/teraz_gramy/teraz_gramy_widget.dart

```diff
@@ -20,40 +23,89 @@ class TerazGramyWidget extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    return BlocBuilder<AudioHandlerCubit, AudioHandler?>(
+        builder: (context, audioHandler) {
     final defaultShadowColor = context.colors.highlightRed;
 
     return ColorShadowedCard(
       shadowColor: shadowColor ?? defaultShadowColor,
       child: ImageWithOverlay(
         imageBuilder: Image.asset,
         thumbnailPath: 'assets/teraz_gramy_background.webp',
-        titleOverlay: RichText(
-          text: TextSpan(
-            children: [
-              TextSpan(
-                text: 'Teraz gramy \n',
-                style: context.textStyles.textPlayer.copyWith(
-                  color: context.colors.highlightGreen,
+        titleOverlay: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: switch (audioHandler) {
+            null =>
+            [
+              RichText(
+                text: TextSpan(
+                  children: [
+                    TextSpan(
+                      text: 'Teraz gramy \n',
+                      style: context.textStyles.textPlayer.copyWith(
+                        color: context.colors.highlightGreen,
+                      ),
+                    ),
+                    TextSpan(
+                      text: 'No stream title',
+                      style: context.textStyles.textMedium.copyWith(
+                        height: 1.5,
+                      ),
+                    ),
+                  ],
                 ),
               ),
-              TextSpan(
-                text: 'RDS streamu',
-                // TODO: Place for the stream RDS
+            ],
+
+            _ =>
+            [RichText(
+              text: TextSpan(
+                children: [
+                  TextSpan(
+                    text: 'Teraz gramy',
+                    style: context.textStyles.textPlayer.copyWith(
+                      color: context.colors.highlightGreen,
+                    ),
+                  ),
+                ],
+              ),
+            ),
+              StreamTitle(
+                audioHandler: audioHandler,
+                width: MediaQuery.of(context).size.width,
                 style: context.textStyles.textMedium.copyWith(
                   height: 1.5,
                 ),
               ),
             ],
-          ),
+          },
```

Comment:
Also here - after the aforementioned correction, both versions have the same text above - "Teraz gramy", so this should really only be a switch between the Text('No stream title') and the StreamTitle widgets, kinda like so:

```dart
    children: [
        Text(
            'Teraz gramy',
             style: context.textStyles.textPlayer.copyWith(
               color: context.colors.highlightGreen,
             ),
           ),
           switch (audioHandler) {
             null => Text(
                 'No stream title',
                 style: context.textStyles.textMedium.copyWith(
                   height: 1.5,
                 ),
               ),
             _ => StreamTitle(
                 audioHandler: audioHandler,
                 width: MediaQuery.of(context).size.width,
                 style: context.textStyles.textMedium.copyWith(
                   height: 1.5,
                 ),
               ),
           },
         ],
```